src/lib.rs

```diff
@@ -237,6 +237,18 @@ mod test {
                 .is_empty());
         }
     }
+
+    #[test]
+    fn check_long_os_version() {
```

Comment:
Could you put this check in `check_system_info` instead please? Since we already have the two other version checks in it, it would make more sense to group this one as well.